pkg/compiler/compiler.go

```diff
@@ -67,6 +74,13 @@ func (c *Compiler) compileMain(binaryName, path string) error {
 }
 
 func (c *Compiler) compileTests(binaryName, path string) error {
+	// do not build if binary is already there
+	binPath := filepath.Join(path, binaryName)
+	if executebleExists(binPath) {
+		c.logger.Log("info", "test binary exists, not building")
+		return nil
```

Comment:
Does `Run()` behave well with `nil` in the bundle to execute? I would be concerned here.

Reply:
Indeed, that's the expected behavior, a `Task` returns `error` https://github.com/giantswarm/e2e-harness/blob/master/pkg/tasks/tasks.go#L6 and if the error returned is nil `Run` keeps processing tasks https://github.com/giantswarm/e2e-harness/blob/master/pkg/tasks/tasks.go#L12.

Reply:
`err == nil` is always good :D

Reply:
Yes, I messed up here - probably shouldn't be doing reviews so late in the evening :D